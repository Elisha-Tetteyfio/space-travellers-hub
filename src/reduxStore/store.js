import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});

export default store;
