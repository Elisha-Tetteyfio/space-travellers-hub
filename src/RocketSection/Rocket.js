/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Rocket.module.css';
import { makeReservation, cancelReservation } from '../reduxStore/rockets';

function Rocket({
  title, maneno, Ximg, id, reservation,
}) {
  const dispatch = useDispatch();

  const update = () => {
    dispatch(makeReservation(id));
  };

  const cancel = () => {
    dispatch(cancelReservation(id));
  };

  return (
    <li className={styles.allInfo}>
      <img src={Ximg} alt="Rocket img" className={styles.img} />
      <div className={styles.rocketInfo}>
        <h2 className={styles.rocketTitle}>{title}</h2>
        <p>{reservation ? <p className={styles.Badge}>Reserved</p> : null } {maneno}</p>
        {reservation ? <button className={styles.cancel} onClick={cancel} type="button">Cancel Reservation</button> : <button className={styles.make} onClick={update} type="button">Reserve Rocket</button>}
      </div>
    </li>
  );
}

Rocket.propTypes = {
  title: PropTypes.string.isRequired,
  maneno: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  Ximg: PropTypes.array.isRequired,
  id: PropTypes.number.isRequired,
  reservation: PropTypes.bool,
};

Rocket.defaultProps = {
  reservation: false,
  maneno: 'Space X',
};

export default Rocket;
