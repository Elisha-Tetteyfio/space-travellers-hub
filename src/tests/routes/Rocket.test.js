import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reduxStore/store';
import Rocket from '../../RocketSection/Rocket';

describe('Testing the Rocket component', () => {
  it('test Rocket', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rocket />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
