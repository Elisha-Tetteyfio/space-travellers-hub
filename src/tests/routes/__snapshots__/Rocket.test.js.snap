// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Rocket component test Rocket 1`] = `
<li
  className="allInfo"
>
  <img
    alt="Rocket img"
    className="img"
  />
  <div
    className="rocketInfo"
  >
    <h2
      className="rocketTitle"
    />
    <p>
       
    </p>
    <button
      className="make"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</li>
`;
